File: scripts/generate_blog.rb
require 'net/http'
require 'json'
require 'fileutils'

# Configuration
GEMINI_API_URL = 'https://api.gemini.com/v2'
BLOG_DIR = 'blog'
IMAGES_DIR = 'assets/images'

# Fetch data from Gemini API
def fetch_gemini_data
  uri = URI("#{GEMINI_API_URL}/data")
  response = Net::HTTP.get(uri)
  JSON.parse(response)
end

# Generate markdown content for a blog post
def generate_markdown(data)
  <<~MARKDOWN
    # #{data['title']}

    #{data['content']}

    ![Image](#{IMAGES_DIR}/#{data['image']})
  MARKDOWN
end

# Save markdown content to a file
def save_markdown(filename, content)
  FileUtils.mkdir_p(BLOG_DIR)
  File.write(File.join(BLOG_DIR, filename), content)
end

# Main function to generate blog posts
def generate_blog_posts
  data = fetch_gemini_data
  data.each do |post|
    markdown = generate_markdown(post)
    save_markdown("#{post['title'].downcase.gsub(' ', '_')}.md", markdown)
  end
end

# Run the script
generate_blog_posts

File: scripts/generate_blog.js
const {
  GoogleGenerativeAI,
  HarmCategory,
  HarmBlockThreshold,
} = require("@google/generative-ai");
const fs = require('fs');
const path = require('path');

const apiKey = process.env.GEMINI_API_KEY;
const genAI = new GoogleGenerativeAI(apiKey);

const model = genAI.getGenerativeModel({
  model: "gemini-2.0-flash-exp",
});

const generationConfig = {
  temperature: 1,
  topP: 0.95,
  topK: 40,
  maxOutputTokens: 8192,
  responseMimeType: "text/plain",
};

async function fetchGeminiData() {
  const chatSession = model.startChat({
    generationConfig,
    history: [],
  });

  const result = await chatSession.sendMessage("Fetch technical blog data");
  return JSON.parse(result.response.text());
}

function generateMarkdown(data) {
  return `
    # ${data.title}

    ${data.content}

    ![Image](assets/images/${data.image})
  `;
}

function saveMarkdown(filename, content) {
  const blogDir = path.join(__dirname, '..', 'blog');
  if (!fs.existsSync(blogDir)) {
    fs.mkdirSync(blogDir);
  }
  fs.writeFileSync(path.join(blogDir, filename), content);
}

async function generateBlogPosts() {
  const data = await fetchGeminiData();
  data.forEach(post => {
    const markdown = generateMarkdown(post);
    saveMarkdown(`${post.title.toLowerCase().replace(/ /g, '_')}.md`, markdown);
  });
}

generateBlogPosts();
